assets/js/job-submission.js

```diff
@@ -3,8 +3,63 @@ jQuery(document).ready(function($) {
 		$(this).closest( '.job-manager-uploaded-file' ).remove();
 		return false;
 	});
-	$( document.body ).on( 'submit', '.job-manager-form:not(.prevent-spinner-behavior)', function() {
+	$( document.body ).on( 'submit', '.job-manager-form:not(.prevent-spinner-behavior)', function( event ) {
+		// Validate the category field if present
+		var jobCategory = $( '#job_category' );
```

Comment:
It is possible that the field has been filtered to not be required. 

Template file [form-fields/term-multiselect-field.php](https://github.com/Automattic/wp-job-manager/blob/332601223930d0a4e26b327700f754c705421e50/templates/form-fields/term-multiselect-field.php) would need to be updated to pass `required` from field settings to `job_manager_dropdown_categories()`. `job_manager_dropdown_categories()` would need to be updated to _probably_ add a class to the field. Adding the HTML5 `required` prop actually seems to do the HTML5 validation to the hidden field (on Chrome, at least), but it is positioned in a strange place (where the hidden field sits above the `select2` field). I'm not sure if this is cross browser (it seems like you had to make the `<textarea>` to be visible to get it to show for you?).

Reply:
I've added some checks that the `required-field` class is present. If not there it will skip the validation entirely. 

I modified the test code you sent me to also allow the description field to be optional if you want to test with it:
`add_filter( 'submit_job_form_fields', function( $fields ) {
		  $fields['job']['job_category']['required'] = false;
		  $fields['job']['job_description']['required'] = false;
		  if ( isset( $fields['job']['job_type'] ) ) {
		     $fields['job']['job_type']['required'] = false;
		     $fields['job']['job_description']['required'] = false;
		  }
		  return $fields;
} );`